File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int fl=word1.length();
        int sl=word2.length();
        int minLen=Math.min(sl,fl);
        StringBuilder sb= new StringBuilder();
        int i=0;
        while (i<minLen){
             sb.append(word1.charAt(i));
             sb.append(word2.charAt(i));
            i++;
        }
        int ffl=fl-minLen;
        int ssl=sl-minLen;
        if(ffl>-1){
            for (int j = i; j < fl; j++) {
                sb.append(word1.charAt(j));
                
            }
        }
        if(ssl>-1){
            for (int j = i; j < sl; j++) {
                sb.append(word2.charAt(j));

            }
        }


        return sb.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        a1=list(word1)
        a2=list(word2)
        ans=[]
        for i in range(max(len(a1),len(a2))):
            if i<len(a1):
                ans.append(a1[i])
            if i<len(a2):
                ans.append(a2[i])
        return "".join(ans)
            